lib/metadata_service.js

```diff
@@ -1,6 +1,8 @@
 var AWS = require('./core');
 require('./http');
 var inherit = AWS.util.inherit;
+var getMetadataServiceEndpoint = require('./metadata_service/get_metadata_service_endpoint');
+var URL = require('url').URL;
```

Comment:
`URL` is added since v6.13. You can try using [urlParse](https://github.com/aws/aws-sdk-js/blob/f1c69ef7606a53f143f70c9da5f6e9737be4b060/lib/util.js#L59)

Reply:
Done in [`d274648` (#3853)](https://github.com/aws/aws-sdk-js/pull/3853/commits/d2746480fd302a744f66731ec06a2109a374ceec)

Reply:
I'd to revert using WHATWG URL, as url.parse wasn't throwing errors for invalid URLs in testing.
Revert commit: [`9879be7` (#3853)](https://github.com/aws/aws-sdk-js/pull/3853/commits/9879be73790a6e0f5d6ad804906813782e25dd26)

Although URL was introduced in v6.13, infinitesimal number of AWS SDK for JavaScript (v2) customers are using it and we're deprecating it on October 30, 2021